stdlib/Libdl/test/runtests.jl

```diff
@@ -222,4 +222,27 @@ let dl = C_NULL
     @test_skip !Libdl.dlclose(dl)   # Syscall doesn't fail on Win32
 end
 
+# test @executable_path expansion
+mktempdir() do dir
+    # Create a `libdcalltest` in a directory that is not on our load path
+    src_path = joinpath(private_libdir, "libccalltest.$(Libdl.dlext)")
+    dst_path = joinpath(dir, "libdcalltest.$(Libdl.dlext)")
+    cp(src_path, dst_path)
+
+    # Add this temporary directory to our load path, using `@executable_path` to do so.
+    push!(Base.DL_LOAD_PATH, joinpath("@executable_path", relpath(dir, Sys.BINDIR)))
```

Comment:
Thanks for the test case, but it's actually the branch without `@executable_path` that is relevant here, i.e.
```suggestion
    push!(Base.DL_LOAD_PATH, dir)
```
But it might be useful to include this test as is and another one without `@executable_path`?

Reply:
I've added another test after this one where the last entry in `Base.DL_LOAD_PATH` is replaced with `dir`. Also, I first add an overly long entry to `Base.DL_LOAD_PATH` which makes these tests segfault without this PR. (I couldn't trigger an error due to the missing zero termination; probably we're lucky and the buffer is usually zeroed out when we get to use it.)

@staticfloat does this look good to you?

Reply:
Ah, I had missed this test (before and after my change) fails on Windows. Any idea?

Reply:
Aha, figured it out.  It's because on windows our temporary directory and our executable are on different filesystems.  Darn.

Reply:
Ah, great, that might have taken me quite a bit of time to track down...